.github/workflows/e2e.yaml

```diff
@@ -40,7 +42,39 @@ jobs:
           aws_access_key_id: ${{ secrets.AWS_ACCESS_KEY_ID }}
           aws_secret_access_key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
           aws_ssh_key: ${{ secrets.AWS_SSH_KEY }}
-          holodeck_config: "tests/e2e/holodeck.yaml"
+          holodeck_config: "tests/e2e/infra/aws.yaml"
+      - name: Run e2e tests
+        env:
+          KUBECONFIG: ${{ github.workspace }}/kubeconfig
+          E2E_IMAGE_REPO: ghcr.io/nvidia/k8s-device-plugin
+          E2E_IMAGE_TAG: ${COMMIT_SHORT_SHA}-ubuntu22.04
+          LOG_ARTIFACTS: ${{ github.workspace }}/e2e_logs
+        run: |
+          make -f tests/e2e/Makefile e2e-test
+      - name: Archive test logs
+        if: ${{ failure() }}
+        uses: actions/upload-artifact@v4
+        with:
+          name: e2e-test-logs
+          path: ./e2e_logs/
+          retention-days: 15
+  nightly-e2e:
+    runs-on: pdx01-arc-runners
+    if: ${{ github.event.workflow_run.event == 'schedule' }}
+    steps:
+      - uses: actions/checkout@v4
+        name: Check out code
+      - name: Calculate build vars
+        id: vars
+        run: |
+          echo "COMMIT_SHORT_SHA=${GITHUB_SHA:0:8}" >> $GITHUB_ENV
+          echo "HOLODECK_VCENTER_USERNAME=${HOLODECK_VCENTER_USERNAME}" >> $GITHUB_ENV
+          echo "HOLODECK_VCENTER_PASSWORD=${HOLODECK_VCENTER_PASSWORD}" >> $GITHUB_ENV
```

Comment:
Doesnt' this mean it's already in the env?

Reply:
since `holodeck` is a container action, we need to pass the env to the container env.

Reply:
For the aws kas we use `with` instead of setting the env. Why the difference?

Reply:
Because these are ENV VARS at the runner level, `with` is to pass GitHub stored secrets.  Since the idea for this Staging job is to test the internal runners, this marks a difference.

Reply:
I don't think that really explains why we can't do:
```
with:
  vcenter_username: ${{ secrets.VCENTER_USERNAME }}
  vcenter_password: ${{ secrets.VCENTER_PASSWORD }}
  vsphere_ssh_key: ${{ secrets.VSPHERE_SSH_KEY }}
```

It is my understanding that these are converted to `INPUT_*` environment variables in any case and will be accessible to holodeck.

Maybe I'm missing something though.

Reply:
GitHub don't inject variables during a non `Post` job, or a job that is forked. 
This Staging job is to test that we can run e2e during `pull` events, meaning we won't have secrets injected by GitHub in the default way, but we are doing secret injection via GitHub ARC config

Reply:
But this job is:
```
if: ${{ github.event.workflow_run.conclusion == 'success' }} && ${{ github.event.workflow_run.event == 'push' }}
```
which is not a PR trigger, or am I missing something?

Reply:
Moved to it's own workflow file, and now runs on PR's (if the image workflow passes)